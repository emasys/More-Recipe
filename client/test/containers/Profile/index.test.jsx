import React from 'react';
import { shallow, mount } from 'enzyme';
import Router from 'react-mock-router';

// Component
import {
  Profile,
  mapStateToProps
} from '../../../src/containers/Profile/index';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

jest.mock('../../../src/containers/Profile/helper');
jest.mock('../../../src/components/CatalogList');
jest.mock('../../../src/components/Navbar');

describe('Test suite for profile page component', () => {
  const props = {
    getUserInfo: jest.fn(),
    getUserRecipes: jest.fn(),
    clearRecipes: jest.fn(),
    updateUser: jest.fn(),
    uploadImg: jest.fn(),
    match: { params: { id: 1 } },
    user: [],
    userInfo: fakeStore.user.userInfo,
    recipes: {},
    auth: fakeStore.user,
    count: 1,
    delRecipe: jest.fn()
  };
  beforeEach(() => {
    wrapper = mount(<Router>
      <Profile {...props} />
    </Router>);
  });
  it('Should render with props', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should view more recipe', () => {
    const viewMore = jest.spyOn(Profile.prototype, 'viewMore');

    const app = mount(<Router><Profile {...props} /></Router>);

    app.find('button#viewMore').simulate('click', {
      preventDefault: () => {}
    });

    expect(viewMore).toHaveBeenCalled();
  });

  it('should delete a recipe', () => {
    const confirmDelete = jest.spyOn(Profile.prototype, 'confirmDelete');

    const app = mount(<Router><Profile {...props} /></Router>);

    app.find('button#confirmDelete').simulate('click', {
      preventDefault: () => {}
    });

    expect(confirmDelete).toHaveBeenCalled();
  });

  it('Should complete form and submit', () => {
    const showForm = jest.spyOn(Profile.prototype, 'showForm');
    const edit = jest.spyOn(Profile.prototype, 'editProfile');
    const blur = jest.spyOn(Profile.prototype, 'onFocus');
    const focus = jest.spyOn(Profile.prototype, 'onBlur');
    const hoverIn = jest.spyOn(Profile.prototype, 'hoverIn');
    const hoverOut = jest.spyOn(Profile.prototype, 'hoverOut');
    const goBack = jest.spyOn(Profile.prototype, 'goBack');


    const app = mount(<Router><Profile {...props} /></Router>);

    app.find('button#edit-profile').simulate('click', {
      preventDefault: () => {}
    });

    app.find('button#goBack').simulate('click', {
      preventDefault: () => {}
    });

    expect(goBack).toHaveBeenCalled();

    app.find('button#edit-profile').simulate('click', {
      preventDefault: () => {}
    });

    expect(showForm).toHaveBeenCalled();

    app.find("input[name='firstName']").simulate('change', { target: { value: 'emmy' } });
    app.find("input[name='firstName']").simulate('blur');
    app.find("input[name='firstName']").simulate('focus');
    app.find("input[name='lastName']").simulate('change', { target: { value: 'endy' } });
    app.find("TextareaAutosize[name='bio']")
      .simulate('change', { target: { value: 'Test writer' } });

    app.find('.img-wrapper').simulate('mouseEnter');
    app.find('.img-wrapper').simulate('mouseLeave');

    expect(hoverOut).toHaveBeenCalled();
    expect(hoverIn).toHaveBeenCalled();

    app.find('form').simulate('submit');

    expect(edit).toHaveBeenCalled();
    expect(blur).toHaveBeenCalled();
    expect(focus).toHaveBeenCalled();
  });


  it('should call componentWillReceiveProps and unmount', () => {
    const app = shallow(<Profile {...props} />);
    const instance = app.instance();
    const compWRP = jest.spyOn(instance, 'componentWillReceiveProps');
    const newProps = {
      count: 1
    };
    const nextProps = {
      count: 20
    };

    // expect(props.clearRecipes).toHaveBeenCalled();

    app.setProps(newProps);
    expect(compWRP).toHaveBeenCalled();
    app.setProps(nextProps);
    expect(compWRP).toHaveBeenCalled();

    app.unmount();
  });

  it('should test mapStateToProps', () => {
    const initialState = {
      recipes: { userRecipes: [], userInfo: {}, userRecipesCount: 0 },
      user: { userInfo: {}, updateUser: {} }
    };
    expect(mapStateToProps(initialState).count).toEqual(0);
    expect(mapStateToProps(initialState).recipes).toEqual([]);
    expect(mapStateToProps(initialState).userInfo).toEqual({});
    expect(mapStateToProps(initialState).auth)
      .toEqual({ userInfo: {}, updateUser: {} });
    expect(mapStateToProps(initialState).updateUser).toEqual({});
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
