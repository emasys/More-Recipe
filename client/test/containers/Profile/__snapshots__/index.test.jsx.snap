// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for profile page component should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Profile
    auth={
      Object {
        "allUsers": Object {
          "success": true,
          "users": Array [
            Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "createdAt": "2018-03-13T05:04:37.443Z",
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
              "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
              "updatedAt": "2018-03-13T05:04:37.443Z",
            },
          ],
        },
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
        "message": null,
        "userInfo": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": "Nigeria",
            "email": "emasysnd@gmail.com",
            "firstName": "emmy",
            "id": 1,
            "lastName": "endy",
            "moniker": "admin",
          },
          "success": true,
        },
        "userProfile": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": null,
            "email": "emasysnd@gmail.com",
            "firstName": null,
            "id": 1,
            "lastName": null,
            "moniker": "admin",
          },
          "success": true,
        },
      }
    }
    clearRecipes={[Function]}
    count={1}
    delRecipe={[Function]}
    getUserInfo={[Function]}
    getUserRecipes={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    recipes={Object {}}
    updateUser={[Function]}
    uploadImg={[Function]}
    user={Array []}
    userInfo={
      Object {
        "data": Object {
          "avatar": null,
          "bio": null,
          "country": "Nigeria",
          "email": "emasysnd@gmail.com",
          "firstName": "emmy",
          "id": 1,
          "lastName": "endy",
          "moniker": "admin",
        },
        "success": true,
      }
    }
  >
    <div>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={<DefaultCloseButton />}
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      >
        <div />
      </ToastContainer>
      <section
        className="container-fluid profile catalog-wrapper-full-grid mt-70"
      >
        <div
          className="row justify-content-center"
        >
          <DeleteModal
            auth={
              Object {
                "allUsers": Object {
                  "success": true,
                  "users": Array [
                    Object {
                      "avatar": null,
                      "bio": null,
                      "country": null,
                      "createdAt": "2018-03-13T05:04:37.443Z",
                      "email": "emasysnd@gmail.com",
                      "firstName": null,
                      "id": 1,
                      "lastName": null,
                      "moniker": "admin",
                      "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                      "updatedAt": "2018-03-13T05:04:37.443Z",
                    },
                  ],
                },
                "authInfo": Object {
                  "isLoggedIn": true,
                  "userId": 1,
                  "username": "admin",
                },
                "isLoggedIn": true,
                "message": null,
                "userInfo": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": "Nigeria",
                    "email": "emasysnd@gmail.com",
                    "firstName": "emmy",
                    "id": 1,
                    "lastName": "endy",
                    "moniker": "admin",
                  },
                  "success": true,
                },
                "userProfile": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": null,
                    "email": "emasysnd@gmail.com",
                    "firstName": null,
                    "id": 1,
                    "lastName": null,
                    "moniker": "admin",
                  },
                  "success": true,
                },
              }
            }
            clearRecipes={[Function]}
            confirmDelete={[Function]}
            count={1}
            delRecipe={[Function]}
            getUserInfo={[Function]}
            getUserRecipes={[Function]}
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
            recipes={Object {}}
            updateUser={[Function]}
            uploadImg={[Function]}
            user={Array []}
            userInfo={
              Object {
                "data": Object {
                  "avatar": null,
                  "bio": null,
                  "country": "Nigeria",
                  "email": "emasysnd@gmail.com",
                  "firstName": "emmy",
                  "id": 1,
                  "lastName": "endy",
                  "moniker": "admin",
                },
                "success": true,
              }
            }
          >
            <div
              aria-hidden="true"
              aria-labelledby="deleteModalTitle"
              className="modal fade"
              id="deleteModal"
              role="dialog"
              tabIndex="-1"
            >
              <div
                className="modal-dialog modal-dialog-centered"
                role="document"
              >
                <div
                  className="modal-content"
                >
                  <div
                    className="modal-header"
                  >
                    <h5
                      className="modal-title"
                      id="exampleModalLongTitle"
                    >
                      Delete Item
                    </h5>
                    <button
                      aria-label="Close"
                      className="close"
                      data-dismiss="modal"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        Ã—
                      </span>
                    </button>
                  </div>
                  <div
                    className="modal-body"
                  >
                    Are you sure you want to continue with this action?
                  </div>
                  <div
                    className="modal-footer"
                  >
                    <button
                      className="btn btn-secondary btn-lg"
                      data-dismiss="modal"
                      type="button"
                    >
                      No
                    </button>
                    <button
                      className="btn btn-danger btn-lg"
                      data-dismiss="modal"
                      id="confirmDelete"
                      onClick={[Function]}
                      type="button"
                    >
                      Yes
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </DeleteModal>
          <UserInfo
            auth={
              Object {
                "allUsers": Object {
                  "success": true,
                  "users": Array [
                    Object {
                      "avatar": null,
                      "bio": null,
                      "country": null,
                      "createdAt": "2018-03-13T05:04:37.443Z",
                      "email": "emasysnd@gmail.com",
                      "firstName": null,
                      "id": 1,
                      "lastName": null,
                      "moniker": "admin",
                      "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                      "updatedAt": "2018-03-13T05:04:37.443Z",
                    },
                  ],
                },
                "authInfo": Object {
                  "isLoggedIn": true,
                  "userId": 1,
                  "username": "admin",
                },
                "isLoggedIn": true,
                "message": null,
                "userInfo": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": "Nigeria",
                    "email": "emasysnd@gmail.com",
                    "firstName": "emmy",
                    "id": 1,
                    "lastName": "endy",
                    "moniker": "admin",
                  },
                  "success": true,
                },
                "userProfile": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": null,
                    "email": "emasysnd@gmail.com",
                    "firstName": null,
                    "id": 1,
                    "lastName": null,
                    "moniker": "admin",
                  },
                  "success": true,
                },
              }
            }
            changeDp={[Function]}
            count={1}
            data={
              Object {
                "data": Object {
                  "id": "",
                },
              }
            }
            handleDrop={[Function]}
            handleImg={[Function]}
            hoverIn={[Function]}
            hoverOut={[Function]}
            notify={[Function]}
            showForm={[Function]}
            state={
              Object {
                "bio": "",
                "country": "",
                "display": "d-block",
                "edit": false,
                "files": "",
                "firstName": "",
                "lastName": "",
                "limit": 6,
                "offset": 6,
                "open": false,
                "preview": "",
                "recipes": Array [],
                "save": "d-none",
                "showMore": true,
                "status": "fade",
                "successMsg": false,
                "view": false,
              }
            }
          >
            <div
              className="col-lg-2 col-md-3 col-sm-6 mb-10 col-12"
            >
              <div
                className="img-wrapper"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className=" changeDp hovered  fade"
                  onClick={[Function]}
                >
                  <Dropzone
                    accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png"
                    className=" p-10 text-center text-light dropzone-dp"
                    disableClick={false}
                    disablePreview={false}
                    disabled={false}
                    maxSize={Infinity}
                    minSize={0}
                    multiple={false}
                    onDrop={[Function]}
                    preventDropOnDocument={true}
                  >
                    <div
                      aria-disabled={false}
                      className=" p-10 text-center text-light dropzone-dp"
                      onClick={[Function]}
                      onDragEnter={[Function]}
                      onDragLeave={[Function]}
                      onDragOver={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      style={Object {}}
                    >
                      click to change profile picture
                      <input
                        accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png"
                        autoComplete="off"
                        disabled={false}
                        multiple={false}
                        onChange={[Function]}
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                        type="file"
                      />
                    </div>
                  </Dropzone>
                </div>
                <img
                  alt="avi"
                  className="img-fluid rounded mb-3"
                  src="https://res.cloudinary.com/emasys/image/upload/v1512284260/mejglndrmulkudy2feug.png"
                />
              </div>
              <div
                className="bg-light rounded p-10 profile-wrapper"
              >
                <h2
                  className="mb-10 bolder"
                />
                <div>
                  <p />
                  <hr />
                </div>
                <small>
                  <i
                    aria-hidden="true"
                    className="fa fa-envelope"
                  />
                   
                </small>
                <div>
                  <button
                    className="btn btn-success btn-lg d-none"
                    id="upload-dp"
                    onClick={[Function]}
                  >
                    Upload Image
                  </button>
                </div>
                <div
                  className="mt-5"
                >
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/manageUsers"
                  >
                    <a
                      className="btn btn-lg btn-light"
                      onClick={[Function]}
                    >
                       
                      Manage Users
                    </a>
                  </Link>
                </div>
                <div
                  className="mt-5"
                >
                  <button
                    className="btn btn-dark"
                    id="edit-profile"
                    onClick={[Function]}
                  >
                    Edit profile
                  </button>
                </div>
              </div>
            </div>
          </UserInfo>
          <div
            className="col-lg-10 col-md-9 col-sm-6 col-12 recipe-lists"
          >
            <div
              className="clearfix"
            >
              <h2
                className=" float-left clearfix"
              >
                Recipes
                 
                <span
                  className="badge badge-dark"
                  data-tip="Total number of recipes added"
                >
                  1
                </span>
              </h2>
              <Link
                className="btn btn-dark float-right clearfix"
                replace={false}
                role="button"
                to="/new"
              >
                <a
                  className="btn btn-dark float-right clearfix"
                  onClick={[Function]}
                  role="button"
                >
                  Add New Recipes
                </a>
              </Link>
            </div>
            <hr />
            <div
              className="row justify-content-center"
            >
              <Connect(CatalogList)
                auth={
                  Object {
                    "allUsers": Object {
                      "success": true,
                      "users": Array [
                        Object {
                          "avatar": null,
                          "bio": null,
                          "country": null,
                          "createdAt": "2018-03-13T05:04:37.443Z",
                          "email": "emasysnd@gmail.com",
                          "firstName": null,
                          "id": 1,
                          "lastName": null,
                          "moniker": "admin",
                          "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                          "updatedAt": "2018-03-13T05:04:37.443Z",
                        },
                      ],
                    },
                    "authInfo": Object {
                      "isLoggedIn": true,
                      "userId": 1,
                      "username": "admin",
                    },
                    "isLoggedIn": true,
                    "message": null,
                    "userInfo": Object {
                      "data": Object {
                        "avatar": null,
                        "bio": null,
                        "country": "Nigeria",
                        "email": "emasysnd@gmail.com",
                        "firstName": "emmy",
                        "id": 1,
                        "lastName": "endy",
                        "moniker": "admin",
                      },
                      "success": true,
                    },
                    "userProfile": Object {
                      "data": Object {
                        "avatar": null,
                        "bio": null,
                        "country": null,
                        "email": "emasysnd@gmail.com",
                        "firstName": null,
                        "id": 1,
                        "lastName": null,
                        "moniker": "admin",
                      },
                      "success": true,
                    },
                  }
                }
                catalog={Object {}}
                clearRecipes={[Function]}
                count={1}
                delRecipe={[Function]}
                deleteRecipe={[Function]}
                getUserInfo={[Function]}
                getUserRecipes={[Function]}
                match={
                  Object {
                    "params": Object {
                      "id": 1,
                    },
                  }
                }
                recipes={Object {}}
                showDeleteBtn={true}
                updateUser={[Function]}
                uploadImg={[Function]}
                user={Array []}
                userInfo={
                  Object {
                    "data": Object {
                      "avatar": null,
                      "bio": null,
                      "country": "Nigeria",
                      "email": "emasysnd@gmail.com",
                      "firstName": "emmy",
                      "id": 1,
                      "lastName": "endy",
                      "moniker": "admin",
                    },
                    "success": true,
                  }
                }
              />
            </div>
            <div
              className="row justify-content-center"
            >
              <button
                className=" btn btn-lg btn-outline-dark text-center"
                id="viewMore"
                onClick={[Function]}
              >
                View more
              </button>
            </div>
          </div>
        </div>
      </section>
      <ReactTooltip
        effect="float"
        insecure={true}
        place="bottom"
        resizeHide={true}
        type="dark"
        wrapper="div"
      >
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
        />
      </ReactTooltip>
    </div>
  </Profile>
</MockRouter>
`;
